mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -4798,6 +4798,7 @@ struct FoldTensorCastPackOp : public OpRewritePattern<PackOp> {
     PackOp newOp = rewriter.create<PackOp>(
         op.getLoc(), newOperands[0], newOperands[1], op.getInnerDimsPos(),
         newMixedTileSizes, op.getPaddingValue(), op.getOuterDimsPerm());
+    newOp->setDiscardableAttrs(op->getDiscardableAttrDictionary());
```

Comment:
I don't see other folding patterns in this dialect propagating external attributes. A change like this should have a comment/rationale at the least to start with.

Reply:
The rationale stated claimed in the comments was that in the prior state, the folders were operating in place, which implicitly preserves these attributes. Regardless of whether this is the best kind of assumption to have, it is an inconsistent and unintended state to have this be different just because this pattern does not do an in place modification. 

Agreed, that should have made it to the commit message.

Reply:
But there are numerous patterns in this file (folding or otherwise) that aren't doing in-place modifications (including tensor.unpack) -- they aren't propagating external attributes either.
```
$ grep replaceOpWithNewOp lib/Dialect/Tensor/IR/TensorOps.cpp | wc -l
24
```
(not even counting create<...> + replaceOp)

```
4682     Value newOp = rewriter.create<tensor::UnPackOp>(
  4683         loc, source, dest, unPackOp.getInnerDimsPos(), unPackOp.getMixedTiles(),
  4684         unPackOp.getOuterDimsPerm());
  4685     rewriter.replaceOpWithNewOp<tensor::CastOp>(
  4686         unPackOp, unPackOp.getResult().getType(), newOp);
  4687     return success();
```

Reply:
I am confused as to what then is the point of external attributes or discardable attributes then. 
I agree this isn't consistent. Like I said it's best effort.

I think the more important thing here is that this doesn't change the semantics of the transformation one bit from upstreams perspective.

Reply:
Discardable attributes are safe to use within a pass, but not across pass boundaries. E.g., I remember we had "filter" attributes in Linalg at some point. (Maybe still have them.) But we had to make sure to drop them again at the end of the pass (if they were still there), so they don't leak across pass boundaries.